spec/token_issuer_spec.rb

```diff
@@ -303,6 +310,32 @@ module CF::UAA
 
   end
 
+  context 'pkce with own code verifier' do
+    let(:options) { {basic_auth: false, code_verifier: 'umoq1e_4XMYXvfHlaO9mSlSI17OKfxnwfR5ZD-oYreFxyn8yQZ-ZHPZfUZ4n3WjY_tkOB_MAisSy4ddqsa6aoTU5ZOcX4ps3de933PczYlC8pZpKL8EQWaDZOnpOyB2W'} }
+
+    it 'calculate code_challenge on existing verifier' do
+      redir_uri = 'http://call.back/uri_path'
+      uri_parts = subject.authcode_uri(redir_uri, 'openid').split('?')
+      code_challenge = subject.get_challenge
+      code_verifier = subject.get_verifier
+      params = Util.decode_form(uri_parts[1])
+      params['code_challenge'].should == code_challenge
+      params['code_challenge_method'].should == 'S256'
+      code_verifier.should == options[:code_verifier]
+      code_challenge.should == 'TAnM2AKGgiQKOC16cRpMdF_55qwmz3B333cq6T18z0s'
+    end
+  end
+
+  context 'no pkce active' do
+      it 'calculate code_challenge on existing verifier' do
```

Comment:
Is this description correct? When there is "no pkce active," do we still "calculate code_challenge on existing verifier"?

Reply:
with uaac I noticed that we should have at the begin an extra switch for PKCE on off.
therefore added this. if you use uaac with a very old uaa maybe that ends in problems. and therefore at least in cf-uaa-lib which is used in many ruby libaries, the default for PKCE is off
The test is for this , means no codes are calculated

Reply:
Yes, understood. I'm just wondering why the `it` block description still says "calculate code_challenge on existing verifier":
```
context 'no pkce active as this is the default' do
      #let(:options) { {use_pkce: false} }
      # by default PKCE is off
      it 'calculate code_challenge on existing verifier' do
```
Seems like you wanna say the opposite: "does not calculate code_challenge"